src/ext/handlersFilter.c

```diff
@@ -0,0 +1,357 @@
+#include <regex.h>
+#include <string.h>
+#include "../lib/defines.h" /* valid include since it brings only RDB_* defines */
+#include "common.h"
+
+struct RdbxFilter {
+    regex_t regex_compiled;
+    int exclude;
+    RdbRes cbReturnValue;
+
+    int regexInitialized;  /* for filter keys */
+    RdbDataType type;      /* for filter types */
+    int dbnum;             /* for filter db */
+};
+
+static void deleteFilterCtx(RdbParser *p, void *data) {
+    RdbxFilter *ctx = (RdbxFilter *) data;
+    if (ctx->regexInitialized) {
+        regfree(&ctx->regex_compiled);
+    }
+    RDB_free(p, ctx);
+}
+
+static RdbDataType opcodeToType(int opcode) {
+    static int init = 0;
+    static int opToType[256];
+
+    if (likely(init))
+        return opToType[opcode];
```

Comment:
My answer to Ozan from earlier commit:
> There is some theoretic risk of two threads that might init the array concurrently. But since both write the same constant values then the result will be just the same.

Reply:
Ok I missed that answer :) I think you assume changes of a thread will be visible to other thread atomically, this is not guaranteed. e.g. a thread might observe `init = 1` but `opToType` is filled with garbage or zeros.

Reply:
I see now that I called memset() at the begining. If it wasn't saw, I would have still argue that there is not even theoretic risk by writing two different threads same constant value to the same place and then read it. 
I will modify it. thanks :)

Reply:
ok, i give up. I will move the table into parser context :)